anydex/wallet/iota/iota_wallet.py

```diff
@@ -0,0 +1,334 @@
+        self.created = self.wallet_exists()
+
+        if self.created:
+            db_seed = self.database.query(DatabaseSeed) \
+                .filter(DatabaseSeed.name.__eq__(self.wallet_name)) \
+                .one()
+            self.seed = Seed(db_seed.seed)
+            self.provider = IotaProvider(testnet=self.testnet, seed=self.seed)
+
+    def create_wallet(self):
+        """
+        Create wallet by creating seed, storing it and setting up API access
+        """
+        if self.created:
+            return fail(RuntimeError(f'Iota wallet with name {self.wallet_name} already exists.'))
+
+        self._logger.info(f'Creating Iota wallet with name {self.wallet_name}')
+
+        # generate random seed and store it in the database as a String instead of TryteString
+        self.seed = Seed.random()
+        self.database.add(DatabaseSeed(name=self.wallet_name, seed=self.seed.__str__()))
+        self.database.commit()
+
+        # initialize connection with API through the provider and get an active non-spent address
+        self.provider = IotaProvider(testnet=self.testnet, seed=self.seed)
+        self.created = True
+        return succeed([])
+
+    def wallet_exists(self) -> bool:
+        """
+        Check whether the wallet has been created or not
+        :return: boolean
+        """
+        return self.database.query(exists().where(DatabaseSeed.name == self.wallet_name)).scalar()
+
+    async def get_address(self):
+        """
+        Return a non-spent address: either old one from the database or a newly generated one
+        :return: a non-spent address
+        """
+        if not self.created:
+            return []
+        # fetch all non-spent transactions from the database
+        address_query = self.database.query(DatabaseAddress)
+        non_spent = address_query.filter(DatabaseAddress.is_spent.is_(False)).all()
+
+        # update the database: check whether any of non-spent addresses became spent
+        for address in non_spent:
+            if await self.provider.is_spent(Address(address.address)):
+                address_query.filter(DatabaseAddress.address.__eq__(address.address)).update({
+                    DatabaseAddress.is_spent: True,
+                })
+        self.database.commit()
+        # if any non spent addresses left in the database, return first one
+        non_spent = self.database.query(DatabaseAddress) \
+            .filter(DatabaseAddress.is_spent.is_(False)) \
+            .all()
+
+        if len(non_spent) > 0:
+            return non_spent[0].address
+        # otherwise generate a new one with the new index and append checksum to it
+        spent_count = self.database.query(DatabaseAddress).count()
+        address = await self.provider.generate_address(index=2*spent_count)
+
+        # store address in the database
+        self.database.add(DatabaseAddress(
+            address=address.__str__(),
```

Comment:
what about using str instead of __str__

Reply:
If I am correct, that's something you ended up discussing during the meeting